generators/server/templates/src/main/java/package/security/_TokenCache.java

```diff
@@ -0,0 +1,155 @@
+<%#
+ Copyright 2013-2017 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://jhipster.github.io/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.security;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+
+/**
+ * Simple time-limited cache for login tokens, necessary to avoid concurrent
+ * requests invalidating one another. It uses a {@link java.util.LinkedHashMap}
+ * to keep the tokens in order of expiration, and a dedicated low-priority purger
+ * thread which periodically removes expired tokens from the map.
+ */
+public class TokenCache<T> {
+<%_ /* TODO Should this class be in the server lib instead? */ _%>
```

Comment:
Yes I think this should go into server lib

Reply:
I guess it depends on whether or not anyone is able to look into using spring caches instead of my simple custom class anytime soon -- because then this class will not be needed at all. But if not, and it is determined that something like this PR could be merged as is in the meantime, then yes it should definitely be in the server lib!